plugins/catalog-backend/src/util/conversion.ts

```diff
@@ -32,10 +32,11 @@ export function locationSpecToMetadataName(location: LocationSpec) {
   return `generated-${hash}`;
 }
 
+/** @public */
 export function locationSpecToLocationEntity(
   location: LocationSpec,
   parentEntity?: Entity,
-): LocationEntityV1alpha1 {
+): LocationEntity {
```

Comment:
Oh this is actually interesting. Wanna have a short discussion about it?

I've been pondering the version-less types, and wondering whether they may actually have been a mistake in the first place. The original idea was that it was convenient to have a type that corresponds to the "latest" good version of a kind, but now I'm not so sure. It's perhaps good for the given instant in time, but in the future when we redefine what this type means (suddenly it's a slightly different `LocationEntityV1`) in _a different package_, it'll have a large area of impact and leading to type errors in peripheral places potentially far from the origins.

Especially for something that _returns_ a type (rather than forgivingly tries to accept many types as input), would it be better to be super explicit about what it outputs? If we change the return type to a `V1`, then the change will be local to this function, the changeset will be on this package with migration instructions for only this function and its usages...

Does that make sense? Again, this is meant as an open question, I'd like to hear what you think :)

Reply:
I will change it back.

I was wondering about it, too. Seeing 
```
export interface LocationEntityV1alpha1 extends Entity {
  apiVersion: 'backstage.io/v1alpha1' | 'backstage.io/v1beta1';
```
as well as 
```
export type {
  LocationEntityV1alpha1 as LocationEntity,
  LocationEntityV1alpha1,
} from './LocationEntityV1alpha1';
```
made me decide to change it then.

But overall, I fully agree with you. Having the specific version will allow for easier evolution and clearer migrations.